lib/tdf3/src/tdf.ts

```diff
@@ -1112,14 +1087,20 @@ export class TDF extends EventEmitter {
         }
 
         const [hash, chunk] = this.chunkMap.entries().next().value;
-        const decryptedSegment = await chunk.decryptedChunk;
+        if (!chunk.decryptedChunk) {
+          await new Promise((resolve) => {
+            chunk._resolve = resolve;
+          });
+        }
+        const decryptedSegment = chunk.decryptedChunk;
 
         controller.enqueue(decryptedSegment.payload.asBuffer());
         progress += chunk.encryptedSegmentSize;
         if (progressHandler) {
           progressHandler(progress);
         }
 
+        delete chunk.decryptedChunk;
```

Comment:
delete operator cause to worse performance ,ance than setting to undefined 
https://medium.com/the-node-js-collection/get-ready-a-new-v8-is-coming-node-js-performance-is-changing-46a63d6da4de#94c8